test/core/security/grpc_tls_credentials_options_test.cc

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h"
+#include "test/core/end2end/data/ssl_test_data.h"
+
+#include <gmock/gmock.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/string_util.h>
+#include <gtest/gtest.h>
+
+namespace testing {
+
+static void SetKeyMaterials(grpc_tls_key_materials_config* config) {
+  grpc_ssl_pem_key_cert_pair** key_cert_pair =
+      static_cast<grpc_ssl_pem_key_cert_pair**>(
+          gpr_zalloc(sizeof(grpc_ssl_pem_key_cert_pair*)));
+  key_cert_pair[0] = static_cast<grpc_ssl_pem_key_cert_pair*>(
+      gpr_zalloc(sizeof(grpc_ssl_pem_key_cert_pair)));
+  key_cert_pair[0]->private_key = gpr_strdup(test_server1_key);
+  key_cert_pair[0]->cert_chain = gpr_strdup(test_server1_cert);
+  grpc_tls_key_materials_config_set_key_materials(
+      config, gpr_strdup(test_root_cert),
+      (const grpc_ssl_pem_key_cert_pair**)key_cert_pair, 1);
+}
```

Comment:
Just want to give you a heads up that the C core ownership semantics are going to be changed([reference](https://github.com/grpc/grpc/issues/20161)). This means C core wouldn't steal the ownership, and we probably have to manually free grpc_ssl_pem_key_cert_pair here in this test. 
The PR is https://github.com/grpc/grpc/pull/22171 (Not ready for review yet).
For now I think we are fine to merge these tests.

Reply:
Whatever you prefer! We can merge this one first and add in the necessary changes in #22171, or vice-versa. I'll let you decide.

Reply:
:+1: , let's merge this PR first :-)